src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -1905,9 +1905,18 @@ internal class PSMethodInvocationConstraints
         internal PSMethodInvocationConstraints(
             Type methodTargetType,
             Type[] parameterTypes)
+            : this(methodTargetType, genericTypeParameters: null, parameterTypes)
         {
-            this.MethodTargetType = methodTargetType;
+        }
+
+        internal PSMethodInvocationConstraints(
+            Type methodTargetType,
+            Type[] genericTypeParameters,
+            Type[] parameterTypes)
+        {
+            MethodTargetType = methodTargetType;
             _parameterTypes = parameterTypes;
+            GenericTypeParameters = genericTypeParameters?.ToArray() ?? Array.Empty<Type>();
```

Comment:
Why do we need `ToArray()`?

Reply:
We don't anymore. I probably forgot to remove it :)